src/storage/heap_file.c

```diff
@@ -24883,9 +24883,10 @@ heap_scan_get_visible_version (THREAD_ENTRY * thread_p, const OID * oid, OID * c
 	}
 
       const bool need_check_visibility = scan_cache->mvcc_snapshot != NULL
-	&& scan_cache->mvcc_snapshot->snapshot_fnc != NULL && !mvcc_is_mvcc_disabled_class (class_oid)
-	&& scan_cache->mvcc_snapshot->snapshot_fnc (thread_p, &mvcc_header,
-						    scan_cache->mvcc_snapshot) != SNAPSHOT_SATISFIED;
+	&& scan_cache->mvcc_snapshot->snapshot_fnc != NULL && class_oid != NULL
```

Comment:
Is this optimization still effective with this additional filter? If it's not, we should take out this part from the `heap_get_visible_version_internal`-> `heap_prepare_get_context`. I hate it, though.
```
 7442   /* Output class_oid if necessary. */
 7443   if (context->class_oid_p != NULL && OID_ISNULL (context->class_oid_p)
 7444       && heap_get_class_oid_from_page (thread_p, context->home_page_watcher.pgptr, context->class_oid_p) != NO_ERROR)
 7445     {
 7446       /* Unexpected. */
 7447       assert_release (false);
 7448       goto error;
 7449     }
```

Reply:
Good. We don't need to do that for this optimization. :)